src/test/rgw/rgw_multi/tests.py

```diff
@@ -1314,3 +1315,976 @@ def test_bucket_creation_time():
+
+    # create objects in bucketA
+    create_objects(zcA, bucketA, objnamesA, content)
+
+    zonegroup_meta_checkpoint(zonegroup)
+    zone_data_checkpoint(zoneB, zoneA)
+
+    # verify that objects from zoneA bucketA are synced to both
+    # bucketA & bucketB in zoneB
+    bucket = get_bucket(zcB, bucketA.name)
+    check_objects_exist(bucket, objnamesA, content)
+
+    bucket = get_bucket(zcB, bucketB.name)
+    check_objects_exist(bucket, objnamesA, content)
+
+    remove_sync_policy_group(c1, "sync-group")
+    return
+
+def test_sync_bucket_level():
+    """
+    test_sync_bucket_level:
+        symmetrical flow but do not enable it at zonegroup level
+        configure another policy at bucketA level and enable it
+
+        verify only bucketA syncs but not other buckets.
+    """
+
+    zonegroup = realm.master_zonegroup()
+    zonegroup_meta_checkpoint(zonegroup)
+
+    zonegroup_conns = ZonegroupConns(zonegroup)
+
+    (zoneA, zoneB) = zonegroup.zones[0:2]
+    (zcA, zcB) = zonegroup_conns.zones[0:2]
+    zones = zoneA.name + "," + zoneB.name
+
+    c1 = zoneA.cluster
+
+    c1.admin(['sync policy get'])
+
+    # create sync-group at zonegroup level but do not enable it
+    create_sync_policy_group(c1, "sync-group")
+    create_sync_group_flow_symmetrical(c1, "sync-group", "sync-flow1", zones)
+    create_sync_group_pipe(c1, "sync-group", "sync-pipe", zones, zones)
+
+    zonegroup.period.update(zoneA, commit=True)
+    get_sync_policy(c1)
+
+    objnames = [ 'obj1', 'obj2' ]
+    content = 'asdasd'
+    buckets = []
+
+    # create bucket in zoneA
+    bucketA = create_zone_bucket(zcA)
+    buckets.append(bucketA)
+
+    # configure sync policy for only bucketA and enable it
+    create_sync_policy_group(c1, "sync-bucket", "allowed", bucketA.name)
+    create_sync_group_flow_symmetrical(c1, "sync-bucket", "sync-flowA", zones, bucketA.name)
+    create_sync_group_pipe(c1, "sync-bucket", "sync-pipe", zones, zones, bucketA.name)
+    set_sync_policy_group_status(c1, "sync-bucket", "enabled", bucketA.name)
+
+    zonegroup.period.update(zoneA, commit=True)
+    get_sync_policy(c1, bucketA.name)
+
+    # create object in bucketA
+    create_object(zcA, bucketA, objnames[0], content)
+
+    # create another bucket and objects in it
+    bucketB = create_zone_bucket(zcA)
+    buckets.append(bucketB)
+    create_object(zcA, bucketB, objnames[1], content)
+
+    zonegroup_meta_checkpoint(zonegroup)
+    zone_data_checkpoint(zoneB, zoneA)
+
+    # verify if bucketA objects are synced
+    bucket = get_bucket(zcB, bucketA.name)
+    check_object_exists(bucket, objnames[0], content)
+
+    # bucketB objects should not be synced
+    bucket = get_bucket(zcB, bucketB.name)
+    check_object_not_exists(bucket, objnames[1])
+
+    remove_sync_policy_group(c1, "sync-bucket", bucketA.name)
+    remove_sync_policy_group(c1, "sync-group")
+    return
+
+def test_sync_bucket_to_different_bucket():
+    """
+    test_sync_bucket_to_different_bucket:
```

Comment:
@yehudasa .. this testcase ( and the following ones below) test the bucket sync policy on the bucket as suggested by you. Could you please check if there is any issue with the configuration here. The sync hasn't happened as expected. Same cases worked when configured on zonegroup.

Reply:
@soumyakoduri it does seem like a bug. See https://github.com/yehudasa/ceph/commit/7247d4d59c4d549492bcdc40b4e6a15610aaaf35 for a fix. In general there are two separate cases: one where the source bucket is different, and one where the dest bucket is different. The first is an easier case because the sync mechanism is pull based. The second case is more difficult and involves a hint system, so that the dest bucket could identify the need to pull data. Both need to be tested (you test here the second case).

Also, you don't need to do a second period update if only the bucket sync policy is modified, only need to perform meta checkpoint. See here: https://github.com/yehudasa/ceph/commit/db3780006b84ea124f62db366700aeea2d91417b

Reply:
> @soumyakoduri it does seem like a bug. See [yehudasa@7247d4d](https://github.com/yehudasa/ceph/commit/7247d4d59c4d549492bcdc40b4e6a15610aaaf35) for a fix. In general there are two separate cases: one where the source bucket is different, and one where the dest bucket is different. The first is an easier case because the sync mechanism is pull based. The second case is more difficult and involves a hint system, so that the dest bucket could identify the need to pull data. Both need to be tested (you test here the second case).
> 

The 2 testcases following this one test both the scenarios.. one sync from multiple buckets and another sync to multiple buckets. Even they had failed. Anyways I shall apply your fix and retest. Thanks!

> Also, you don't need to do a second period update if only the bucket sync policy is modified, only need to perform meta checkpoint. See here: [yehudasa@db37800](https://github.com/yehudasa/ceph/commit/db3780006b84ea124f62db366700aeea2d91417b)

Will address this.

Reply:
Thanks @yehudasa .. I applied your patch and that fixed this test case but the other test case at the end "test_sync_bucket_to_multiple_buckets" similar to this one but with multiple destinations bucket is failing. Please take a look.

Also I have addressed your other comments. Kindly review.